### PR TITLE
feat(sdk): `SlidingSyncRoom.timeline` is no longer a `Arc<MutableVec<_>>`

### PR DESCRIPTION
**Pull request by Hywan**
Prior to this patch, the `SlidingSyncRoom.timeline` field was initially of type `AliveRoomTimeline`. This is a type alias to `Arc<MutableVec<SyncTimelineEvent>>`.

No one can listen to the `timeline` field because it's not public. So the `MutableVec` part is not necessary, hence removing many locks. At this step, `AliveRoomTimeline` can simply be defined as `Arc<Vec<SyncTimelineEvent>>`.

But we can go further… There is no need for an `Arc` here. So `timeline`'s type becomes `Vec<SyncTimelineEvent>`. The type is simple enough to remove its type alias `AliveRoomTimeline`. Indeed, `Vec<SyncTimelineEvent>` is used in other types, like `FrozenSlidingSyncRoom` for example. It's not an issue.

To summarize: No more locks, no more clones.

Edit: This patch also removes the conditional `timeline()` method. The `experimental-timeline` feature is always enabled by `experimental-sliding-sync`. It's dead-code basically.

### CODE REVIEW

File: crates/matrix-sdk/src/sliding_sync.rs [hunk 1]
```diff
@@ -148,7 +146,7 @@ pub struct SlidingSyncRoom {
     is_loading_more: Mutable<bool>,
     is_cold: Arc<AtomicBool>,
     prev_batch: Mutable<Option<String>>,
-    timeline: AliveRoomTimeline,
+    timeline: Vec<SyncTimelineEvent>,
```
Comment (poljar) [inline crates/matrix-sdk/src/sliding_sync.rs:149]: But this will now clone a lot of data when we clone the `SlidingSyncRoom`, are we sure we don't need to have this behind an `Arc`?

Comment (Hywan) [reply crates/matrix-sdk/src/sliding_sync.rs:149]: I'm not sure we are cloning this `SlidingSyncRoom` quite often actually. cc @gnunicorn?

Comment (poljar) [reply crates/matrix-sdk/src/sliding_sync.rs:149]: Well it's not up to us if we clone it or now. We have a public getter which clones the room:

https://github.com/matrix-org/matrix-rust-sdk/blob/c7fe1ddffbf15d502efda97b3bda00501f89afd2/crates/matrix-sdk/src/sliding_sync.rs#L816-L819

So it totally depends on what the user does, but I can imagine users of the SDK having clones of the room in a bunch of places. Furthermore if we now remove the `Arc`, users that picked up a room using the getter won't get changes to the `timeline` field anymore. 

If they then call the `timeline()` method using a `SlidingSyncRoom` that has such stale data, they will get a timeline that has been populated with these stale events.

**Comment by Hywan** [comment]
I misunderstand the code. The existing behaviour allows multiple clone of `SlidingSyncRoom` to be updated. Let's not break it.
